torch/_dynamo/variables/user_defined.py

```diff
@@ -581,4 +581,9 @@ def __init__(self, value, **kwargs):
         assert type(value) is KeyedJaggedTensor
         super().__init__(value, **kwargs)
 
-    # TODO Handle getattr for _length_per_key and _offset_per_key properly.
+    def var_getattr(self, tx, name):
+        source = AttrSource(self.source, name) if self.source else None
+        if source is not None and name in ("_length_per_key", "_offset_per_key"):
+            with torch._dynamo.config.patch(force_unspec_int_unbacked_size_like=True):
```

Comment:
context manager hacky patching on config? broke
context manager hacky patching on instruction translator? woke

Have you seen `with tx.strict_translation_mode`? This feels like a sibling, `with tx.force_blah` - and you can then avoid internal settings on config. 

Otherwise, its not that uncommon to patch into a state like this and then resume tracing (down into an inline, up into a super, etc), so I would be okay with this in spirit :)

Reply:
I also need a separate strategy for when we trace through a cumsum().unbind() call internally, in which case I either have to update torchrec to assert "these are sizes" or slide in those asserts myself... maybe just need to overwrite call_method appropriately or something

Reply:
So I took a look at doing this, and one problem with putting it on `tx` is that we generate inner tx when we inline. But I want this property to inherit through the inner tx's. So this seems like, morally, not quite the right spot to put the knob.

Reply:
I put it in TracingContext instead.